core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -50,7 +50,9 @@ private[spark] sealed trait MapStatus {
 private[spark] object MapStatus {
 
   def apply(loc: BlockManagerId, uncompressedSizes: Array[Long]): MapStatus = {
-    if (uncompressedSizes.length > 2000) {
+    if (uncompressedSizes.length >  Option(SparkEnv.get)
```

Comment:
this should be done once, rather than for every constructor, shouldn't it? might introduce a hot codepath for very large workloads here.

Reply:
btw this creates the impression that this can be modified in runtime, but in practice the executors generate these MapStatuses so they can't really be changed on executors.

Reply:
How about creating a "static" val shuffleMinNumPartsToHighlyCompress for this? Please let me know if this is good for you so I can update it. Thanks.

Reply:
yes would be better to be done once, please file a jira.

Reply:
the only tricky thing is how to write the test cases for this.